client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { useAuth } from '../../context/useAuthContext';
+import { Route, Redirect } from 'react-router-dom';
+type Props = {
+  children: JSX.Element;
+  path: string;
+  exact: boolean;
+};
+const ProtectedRoute: React.FC<Props> = ({ children, ...restOfProps }) => {
+  const { loggedInUser } = useAuth();
+  let isLoggedInAndValid = false;
+
+  if (loggedInUser && loggedInUser.username && loggedInUser.email) {
+    isLoggedInAndValid = true;
+  }
+
+  return <Route {...restOfProps} render={() => (isLoggedInAndValid ? children : <Redirect to="/login" />)} />;
```

Comment:
I wasn't sure what was wrong with this at first, turns out if we provide both a `Component` render method and a `render` render method to a `Route`, `Component` will ignore the `render` render method (more on Route render methods [here](https://reactrouter.com/web/api/Route/route-render-methods)). Since your protection logic is embedded within the `render` prop, your component won't work as intended if there is a `component` prop in your `restOfProps`. Here's a [sandbox](https://codesandbox.io/s/route-render-method-test-xsisb) that demonstrates this. 

Let's instead use conditional rendering to render the appropriate component (e.g. a `Route` or a `Redirect`).